src/app/services/organization-permissions-guard.service.ts

```diff
@@ -4,32 +4,44 @@ import { ActivatedRouteSnapshot, CanActivate, Router } from "@angular/router";
 import { I18nService } from "jslib-common/abstractions/i18n.service";
 import { OrganizationService } from "jslib-common/abstractions/organization.service";
 import { PlatformUtilsService } from "jslib-common/abstractions/platformUtils.service";
+import { Permissions } from "jslib-common/enums/permissions";
 
 @Injectable()
-export class OrganizationGuardService implements CanActivate {
+export class OrganizationPermissionsGuardService implements CanActivate {
   constructor(
+    private organizationService: OrganizationService,
     private router: Router,
     private platformUtilsService: PlatformUtilsService,
-    private i18nService: I18nService,
-    private organizationService: OrganizationService
+    private i18nService: I18nService
   ) {}
 
   async canActivate(route: ActivatedRouteSnapshot) {
     const org = await this.organizationService.get(route.params.organizationId);
     if (org == null) {
-      this.router.navigate(["/"]);
-      return false;
+      return this.cancelNavigation();
     }
+
     if (!org.isOwner && !org.enabled) {
       this.platformUtilsService.showToast(
         "error",
         null,
         this.i18nService.t("organizationIsDisabled")
       );
-      this.router.navigate(["/"]);
-      return false;
+      return this.cancelNavigation();
+    }
+
+    const permissions = route.data == null ? null : (route.data.permissions as Permissions[]);
+    if (!org.hasAnyPermission(permissions)) {
+      this.platformUtilsService.showToast("error", null, this.i18nService.t("accessDenied"));
+
+      return this.cancelNavigation();
     }
 
     return true;
   }
+
+  private cancelNavigation() {
+    this.router.navigate(["/"]);
+    return false;
+  }
```

Comment:
Should we create our own base class for router guards?

Reply:
PR for review: https://github.com/bitwarden/jslib/pull/732